internal/test/wsu/wsu_test.go

```diff
@@ -44,6 +39,35 @@ var (
 	ubi8Image = "registry.access.redhat.com/ubi8/ubi:latest"
 )
 
+type WsuFramework struct {
+	// TestFramework holds the instantiation of test suite being executed. As of now, temp dir is hardcoded.
+	*e2ef.TestFramework
+	// Number of VMs which should use built version of WMCB
+	vmCountWithBuiltWmcb int
+}
+
+// Setup initializes the framework. 'vmCountWithBuildWmcb' indicates how many VMs out of 'vmCount' created would run WSU
+// that uses built version of WMCB. Others will download the latest released version of WMCB as per the cluster version
+func (f *WsuFramework) Setup(vmCount int, credentials []*types.Credentials, skipVMsetup bool) error {
+	f.TestFramework = &e2ef.TestFramework{}
+	// vmCountWithBuiltWmcb should not be greater than vmCount
+	// If vmCount is 2, 1 VM will run WSU that will build WMCB and the other will auto-download the latest WMCB based on
+	// the cluster version
+	f.vmCountWithBuiltWmcb = 1
```

Comment:
You're hardcoding this, any reason for doing so or do you want to handle computing what all VMs need local build version in a later stage? If yes, please add a TODO on how want to do it. Perhaps you can create a story around it as well.

Reply:
Changed comment and added a TODO in the struct to explain my thought process